### ISSUE TITLE
Remove the tree life cycle / spreading / growth

### ISSUE THREAD

**Issue by bunnybot** [widelands/widelands#350]
This is the direct opposite of bug 536511.

I am quite annoyed with the tree spreading feature and I want to bring up some arguments why I think it should be removed again. For each argument, I give some examples and tell a story where it bugged me before. I put this up for discussion as I expect there is some controversy around it.

1) It takes freedom from the map designers and introduces arbitrary randomness which is always a bad idea in a competitive game in my opinion.
I think map designers should decide how a map should look. If I want to block expansion, I build a big forest. If I want to make trees scare, I plant less of them on my map. I do not want to have an area overgrown with trees because the player needed 5 hours to reach that part of the map and I planted a forest there before. Shortly: currently I cannot design map features as I want to; because I have no control how much the forest has grown when the player gets there. Same for dying trees: I might want the player to has not many trees in one location, but when he reached the location, the few trees I planted have likely already died; so he has none. Not my intention.

Story: I played the (new) Atlantean tutorial mission and couldn't expand my territory to the east because there was an inpenetratable forest there. Nasenbaer only planted a few trees in the Editor, but they seeded and grew together. Not his intention.

2) It introduces bugs
In the tutorial campaign, I programed a certain number of buildings to be warped in in the eastern part of the map. If the player took to long to reach this part (and/or randomness played against us) the place was crowded with trees: no place for the roads that I wanted to be constructed there. I hacked some stuff into the scripts to remove the trees where the buildings and roads should be build, but this looks rather awkward. I looked at the map as it was in the editor and planed the city there; the tree spreading destroyed my planing - for no reason at all. Another problem could be a computer player that is inactive for a while and gets activated by a lua trigger: in the meantime, he could be completely surrounded by trees because he had no lumberjacks. He will then die right at the start.

3) It makes balancing of maps harder.
Playing widelands in multiplayer, fairness is very important. Making maps fair is hard work; but one thing that is easy is to make sure that everyone has the same number of stones and trees at his starting location. The tree spreading adds arbitrary randomness to the number of trees: some more might die for one player than for the other.

I feel the feature adds very little: The game becomes more realistic, but there is no added gameplay value to this. The spreading of trees is very hard to see while playing (one only sees trees disappear randomly from time to time and at high speeds, one occasionally sees a new tree popping up) and the only thing the player gets to see from this feature is massive forests as he expands. I also think the randomness is a bad thing. I vote for removing this feature from widelands completely.

------------------------------------
Imported from Launchpad using lp2gh.

 * date created: 2010-10-03T13:31:18Z
 * owner: sirver
 * the launchpad url was https://bugs.launchpad.net/bugs/654078

**Comment by bunnybot** [comment]
(by timo-wingender)
I really like this feature and some randomness in the game. Even not much noticeable for me such small details make widelands more realistic. For me this small details are what makes widelands really interesting.

Of course this feature introduces bugs. But that is true for almost every feature.

About map designers: Actually the tree spreading depends heavily on the terrain. So actually the map designer can influence this.

So I am for keeping this feature.

**Comment by bunnybot** [comment]
(by sirver)
Aah, found it: this was also mentioned in the forums as a problem before: http://wl.widelands.org/forum/topic/230. 

I will wait for some more comments before I respond to you, Timowi, so that the discussion is not getting too narrow before others chimed on in on it.

**Comment by bunnybot** [comment]
(by ludamo)
Maybe you should just implement features to let the map designer decide if they want this feature enabled (or disabled) on a map.

Another way could be to let the designers define areas within a map where they can override the global (enabled|disabled) setting ?

**Comment by bunnybot** [comment]
(by nasenbaer)
Actually I am quite heavy in exam preperations - whoeve, I needed to to write my opinion here as fast as possible ;).

Although I was against the "natural" life cycle at the beginning (when reproduction times where very short), I really vote FOR KEEPING this feature, as it gives the game that much more life!
Please do not missunderstand me SirVer ( I do not want to offend you :) ), but if you get annoyed by the trees, you obviously haven't found a good strategy to keep your forests under control. And that's exactly the point why I like the feature - because it makes "economy strategics" a bit more complex and more natural.

Of course there is no reason, why the implementation shouldn't be tweaked further (perhaps highering the reproduction rates even further or decreasing the affinity of some trees for a certain terrain) but a complete remove would rip out a feature, that make Widelands even cooler!

Btw.: I am not aware of any bug that is based upon the tree life cycle - I have only heared of some people, that wondered about the different behaviour in comparence to Settlers 2 - but THAT is not a bug ;).

So to conclude:
A clear NO!

**Comment by bunnybot** [comment]
(by nasenbaer)
talking of scenarios: it should not be too hard to implement an option to en/disable tree growing and add a lua function like:
wl.Map().tree_cycle({enabled = false})

**Comment by bunnybot** [comment]
(by sirver)
#1:
> About map designers: Actually the tree spreading depends heavily on the terrain. So actually the map designer can influence this.
not very much: the trees will have died or grown extensively when the user reaches the part where you planted your trees. Your influence is very indirect which is imho not desirable in scenarios.
I also do not understand how this makes the game more interesting as you hardly ever notice it in game? In fact, you only notice the results: too much trees or no trees at all when you expand further and further on big maps.

#3: if most people are pro keeping it, something like that needs to be implemented. It is not so easy with the current implementation (through immovable programs as they do not support if/else clauses).

#4: Also to you the question: how does something that you hardly ever notice except for it's undesired longterm effects give the game more life. And about me not having found the correct strategy: this is not about forest inside my terrain (there i never noticed the tree life cycles effect), this is only and completely on big maps when you expand: you have to rip millions of trees or you find an empty area; this just plainly sucks (imho, of course :) ).

One bug related to tree growing is 642897.

To all:
:( does noone else feel that randomness in a game is a bad thing? Have you ever played monopoly and lost because you never hid the good cards in the first round? Randomness and arbitrariness are bad for a competitive game, statistics (and there are already too many in widelands) feel unfair even if they might not be in the long run.

Please convince me as I do not understand your points at this moment in time.

**Comment by bunnybot** [comment]
(by nasenbaer)
1) I notice the tree life cycle quite often - even inside my terrain (e.g. when trees from the other side of a river spread young trees on my side of the river, where I've already removed al trees :) )

2) I disagree - I really like randomness in games! :)
Actually Widelands convinced me, that a game can be very fun, even if you are not the final winner - an especially if randomness is involved, you can easier live with loosing a game and will try to improve your strategies further, to not be beaten by the randomness the next time.

To convince you? Well no idea. This is obviously something without a 100% truth. ;)

So yeah, I am still 100% for keeping it, but if you really feel deadly unhappy with the current state, let me make a proposal :) :
You implement:
* a boolean value in the game class, that can be set via lua and  c++ functions, that checks whether the spread and die commands of trees should be run (which means the same like whether the tree life cylce is active - growing and terrain affinity should of course stay, so the foresters will still work as always)
* loading saving code (for the boolean variable)
You would than have the possibility to disable the tree life cylce in scenarios
And I would implement an option to enable/disable tree life cycles in multiplayer games during my seperation of multiplayer launchgame menu and singleplayer launchgame menu (see the shared kingdom reimplementation bug)

**Comment by bunnybot** [comment]
(by timo-wingender)
I totally agree with Nasenbear at this.

I notice the tree spreading often as well. For me this is an integral part of widelands. This adds much ambience to the game. Seeing the forests growing adds life to the game. This and the animals running around makes it much fun too just watch the game. For me removing the tree spreading is almost as invasive as removing the animals from the map.

It like randomness in games too. Especially if it is not few big random decisions but many small events that do not influence the game much. You can not compare the randomness in widelands with having bad luck in games. The randomness in widelands is not so relevant that it will decide the game. It just adds some uncertainty to some events. And I actually like that.

I do not play widelands to win the game. For me it is important to have fun during the game. For me it is not about competition, winning and statistics.

**Comment by bunnybot** [comment]
(by ludamo)
I've noticed this randomness in my 15h play on the riverlands map most...

i don't think randomly growing trees are a really problem (except for scenario maps where the designer want's to place buildings or other stuff triggered...) - but i can understand SirVer about dying trees which had been placed to block player ...

Maybe for this you could implement something like the *undead* trees, which don't die (because they are already) but also can't just removed by building a road, flag or other stuff on the field so they have to be cut by lumberjacks...

**Comment by bunnybot** [comment]
(by ha-kripo)
Just another player's opinion:
I never noticed the trees outside my territory growing. I follow the trees in my territory as they are my future source of trunks. So if I run short of trunks, I feel good when discovering lots of lush small trees on my land. This and many animations (animals, water, wind, workers!) and sounds give life to the game for me.
I do not find the trees outside the territory that important for the realistic feeling to let them introduce multiplayer inequitiy (even if small) and bugs.

**Comment by bunnybot** [comment]
(by sirver)
#7: Your suggestion is a good compromise, but I fear the feature creep and option explosion much more than wrong game mechanics. I prefer to have one ruleset, even if I do not agree with it. We will see how we can cope with it in scenarios; one can still place stones as they do not reproduce or weather (yet?). We'll handle it one way or another.

#8: I agree that competition is not the spirit of widelands. But I feel that a game that is 100% non arbitrary-random can still be fun for the casual gamer (example: starcraft) and I feel that widelands should venture in this direction and keep the wusel in other effects that do not affect the ruleset of the game. What I mean to say is: If you do not play widelands for the win, you shouldn't mind that the game is fair. But currently, people cannot play a fair game, because the wusel introduces arbitrariness. It just goes in one direction.

I disagree that the current randomness is not so relevant: if your soldier hits first and my second attack misses then your chance to win the battle has been increased by a huge amount. The next fight, it might be different, but this one fight could decide over taking or not taking a military building (Bug arbitrariness in military fights has been discussed in bug 577887 and I have been overruled there). Same with trees: if we both have 3 trees at the beginning, one of mine dies and one of yours reproduce, you have build materials for one building more than I do - and wood is so very scarce at the beginning of the game.

I will be away for a week so do not wonder if I do not answer bug reports in this time. Have a nice week everyone.

**Comment by bunnybot** [comment]
(by hjd)
Small elements of randomness make the maps slightly different if you play them several times. I like to think that noticing some small detail that has changed since the last time I played it makes it more interesting than if I had played the same static map. (Note:small changes)
As others have suggested, one possible solution would be to have the map designer enable/disable the tree spreading when creating the map, based on what best fits the map.
Another approach, though it's probably more complex, would be to adjust how the tree spreading works. Some of the problems described are too many trees, or none at all. In other words, the randomness cause large changes which has a large impact on the game, not like the small details I described above. What if instead of planting new trees and increasing the number indefinitely, the number of trees were bounded. Say for instance you place a small forest consisting of 12 trees somewhere, which the player won't get to for a while. By today's system the player might find anything from an overgrown forest to nothing to anything in between. What if instead you would find 12 trees in that area, but if you played the same map, it might be different trees, or organized differently? Due to how that forest had grown, some trees died, some new have grown. In other words, each time you play the map, you will find more or less the same forests in the areas they are supposed to be, but they will be slightly different from time to time. I'm not exactly sure if this is feasible, and lumberjacks/rangers might cause bugs when the change the total number of trees present in the world. 

"how does something that you hardly ever notice except for it's undesired longterm effects give the game more life." Again, I think this is the small/large change caused by randomness. I think small changes has it's place and makes the game more interesting. Too large changes caused by randomness may result in a frustrating experience and this should of course be avoided. I think we should keep it, but it may need to be tweaked.

**Comment by bunnybot** [comment]
(by nasenbaer)
I still do not see the reason why this should give a big influence on fairness:

First of all the point 
> if we both have 3 trees at the beginning, one of mine dies and one of yours reproduce,
> you have build materials for one building more than I do - and wood is so very scarce
> at the beginning of the game.

seems to be invalid for me. The first dying/spreading takes place after 800 - 2500 seconds (depending on the type of tree) - so let's take the worst case - all 3 trees are birches (idle for 800 seconds) - so you have got 13 minutes and 20 seconds to build up a wood economy until one of the trees possibly die. - that is more than enough time. And once you have a forester, one tree more or less definitely does not matter, as tree reproduction through the forester is quite fast.

Second of all "feature creep":
I have thought about it before I wrote my comment and have thought about it afterwards again. You are right, that mikro options can clutter the game and make it harder for people to look trhough all the options. However this would be a per game option (not a general option), so nothing we add to the options and conf files.

**Comment by bunnybot** [comment]
(by ludamo)
Well, over all, if i have to decide this feature, you will find tons of *pros* and *cons* - so, there should really be options to let the player decide (what they want).

If i don't misunderstood the comments above we have two different types where the player should decide:

1. let the map designer decide (could be important for scenarios)
2. let the player decide just before they start the game

the second one could really confuse players (as more options you have to set before you start a game as less players will start a game ;))

I really don't understand lua but as fast as i understood, it would be a real benefit if it is possible to take control over this feature in lua for scripting in scenarios - especially if scenarios are enabled in multiplayer again.

Am i right, that each map has it's own lua init? If so, and if you provide a lua function to (dis)able this feature - would't this satisfy both sides? (Even if not, consider to enable an init script for each map, even if it's not an scenario)

**Comment by bunnybot** [comment]
(by timo-wingender)
The difference is not as drastically as hjd described it. What is there when the player gets there is mostly a question of time and the terrain. If the terrain affinity is bad the player will find less or no trees there. If the affinity is good for the trees the player may find some trees there or finds a big forest if he comes later. The problem for a scenario/map designer it that it may have changed a lot depending of when the player arrives there. But the general tendency is set by the terrain. On possibility is to use different terrain types there. A small spot with good affinity for the trees and around a lot with bad affinity. The the player will most probably find a small forest on this spot.

**Comment by bunnybot** [comment]
(by sigra)
Creating buildings, flags and roads in scenarios works in the latest build, which has the natural tree lifecycle feature. If that no longer works, it is a recent regression not caused by that feature.

This discussion reminds me of a nice little feature of AoE. Computer players would give bad excuses for resigning when they realized that they could not win [http://www.facebook.com/topic.php?uid=2235841931&topic=6626]:

"All my beginning citizenry were male"

 "A boar killed all my people in the town"

"I accidentally resigned by mistake"

"All the gold I had was fake"

"a wild wolf made off with my best sheep. I am therefor forced to resign"


Now a Widelands computer player could say: "one of my 3 trees died"

**Comment by bunnybot** [comment]
(by ha-kripo)
Not sharing the same opinion of the theoretic perfect game is one thing. It's ok and there will be a way to find a solution that pleases many (if not all).
It‘s another to assume that someone does not at all have this idea of a good game, but the fainthearted will to cover his losing. 
It‘s not about searching excuses for not winning that one thinks about removing a working feature from a game and that one starts a discussion like this.

**Comment by bunnybot** [comment]
(by nasenbaer)
Excuse me Kristin (#17): I read your text in English, translated it two times to German and still do not get the essence of what you wrote. (one of my interpretations is, that you feel offended (or think someone else feels), another feels a bit like you want to offend... so before I write a further comment, I better ask what you really wanted to say with your comment :) )

**Comment by bunnybot** [comment]
(by wolfsteinmetz)
As some may remember I was not at all in favour of the auto.spreading trees, when I started to play WL.
Meanwhile, and after the timings in the conf files have been tweaked, I changed my mind and I am quite fond
of the feature. I'd hate to see it go completely.
Still Sirver's objections are true and valid. Only, personally I regard them as less important for my gameplay.
I would love to see the mapmaker in command of disabling or enabling that feature - ideally even set the
spawn rate separately for certain regions on the map. (remember the named regions concept?)
Those are the people with an idea about how the map should work, and I would think that they will welcome
this additional option.

**Comment by bunnybot** [comment]
(by ha-kripo)
@#18: Sorry, the misunderstanding may be due to my english...
I feel offended by post #16. Because:
The arguments contra the tree life cycle are the bugs, the imbalancing and the shortened control of the map/scenario designer. 
Post #16 implies that these are not the true motives of this bug report. It implies that the reasoning contra the tree life cycle is in reality a search for excuses for bad playing. (At least, I cannot understand it otherwise.)  
I wanted to say: it's ok to have different opinions about these trees, but it's not ok (neither logical) to assume that someone who wants to improve the game has only the motive to conceal his bad playing.

in German (just to prevent misunderstandings):
Die Argumente gegen den Baum-Lebenszyklus sind die Bugs, Ungerechtigkeit und verringerte Kontrolle der Karten/Scenariodesigner.
Post #16 unterstellt, dass das aber nicht die wahren Gründe für diesen Bugreport sind. Er unterstellt, dass die Argumentation gegen den Baum-Lebenszyklus in Wahrheit eine Suche nach Entschuldigungen für schlechtes Spielen sind. (Zumindest kann ichs nicht anders verstehen)
Es ist ok, verschiedene Meinungen zu den Bäumen zu haben, aber es ist nicht ok (und auch nicht logisch), jemandem, der das Spiel verbessern will, zu unterstellen, dass er nur sein schlechtes Spielen vertuschen will.

**Comment by bunnybot** [comment]
(by nasenbaer)
@Kristin (#20): I understand. And (although I did not understand #16 the way you understood it) you are of course right, that we should stay impartial.

However, I think we have heard quite a lot opinions. Most are for keeping the feature + adding the possibility for map creators, to disable tree growing.

So my idea would be:
* change the "remove" command (when trees die) with a new "die" command - if the tree cycle is active it will directly call the "remove" command.
* add a test to "spread" command and only run it, if tree cycle is active.
* add a lua variable for wl.Game() to en/disable the tree cycle.

What do others think?

**Comment by bunnybot** [comment]
(by chuckw20)
I like giving the map designers the option to en/disable the tree spreading
feature.  If disabled, does it make sense to still enable a die/remove
action?  That seems counterproductive for a map intended to be predictable
in this respect.  It would theoretically render any map treeless given
sufficient time.

Just my 2 cents.

**Comment by bunnybot** [comment]
(by nasenbaer)
@Chuck (#22): Perhaps I failed to explain this. Actually I wanted to say, that there should be two commands - the already existing "remove" command (as well used by lumberjacks, once they cut a tree) and a new "die" command, which only calls "remove", if the tree cycle is active.

Hope that cleans it up :)

**Comment by bunnybot** [comment]
(by chuckw20)
@Nasenbaer (#23):  By Jove, I think I've got it!  :)

Thanks!

**Comment by bunnybot** [comment]
(by csirkeee)
I know I'm not active around here recently, but I wanted to chime in :).

I am against making this an option, even for map creators, I think it would be confusing for players if on one map when they reach far areas there is always dense forest, while on other maps, it doesn't happen. (I know tree growth was an incentive for me as a player to start spreading quickly, so I don't have to cut down all the trees. Now how do I know if I need to do that, it it's optional?)

But, I have another "compromise", that at least could solve the "find millions of trees or find an empty area" part (similar solution to HJDs): the tree spreading could be influenced by "crowding", that is, a tree has less chance to start growing in a spot, if there are other trees nearby. How much this factors in could depend on tree type and/or terrain. If this is calibrated well, it could lead to nice sparse forests (for some kinds of trees). Of course, I don't know the tree spreading code, so I don't know how easy this would be to implement, or how bad it would be for performance, but if it could be done well it would both improve playability, while being more realistic (after all, there are sparser forests in real life :) )

(This could also be used to make sure trees don't "die out" for example, if there is no other tree for 6 distance, spreading chance becomes 100%, or something like that )

**Comment by bunnybot** [comment]
(by wolfsteinmetz)
@Kiscsirke: Ah - I had the same criticism initially (player must know how the trees will behave), and then simply forgot to mention the point in my posting  :(

This does not really speak against making the tree growth configurable for map makers. 
An obvious solution for this could be to inform the player.
We may want to discuss how this is done best, but there wll be a need for such an info
in other respects too (like: is seafaring enabled on this map or not)

Your suggestion on the other hand would take away the option for mapmakers  
to slow down a players expansion by trees no matter if he advances fast or slowly.
A thick forrest that once was there at the beginning will have degraded to a sparse 
one that does not block if the player is slow.

>> ...that is, a tree has less chance to start growing in a spot, if there are other trees nearby.

And this behaviour seems somewhat "counternatural" to me, given the way that trees actually spread
by seeds. ;)

**Comment by bunnybot** [comment]
(by nasenbaer)
>>> ...that is, a tree has less chance to start growing in a spot,
>>> if there are other trees nearby. 
> 
> And this behaviour seems somewhat "counternatural" to me,
> given the way that trees actually spread by seeds. ;)

Jepp I completely agree (wanted to write the same before I read #26 ;) )! The way it is now seems quite natural to me and influencing this as map creator is still possible via terrain placement (as mentioned above).

So the question is just:
Do we implement a feature to disable the tree cycle? And if yes:
* only for scenarios (via lua)?
* as (by the map creator) hardcoded map variable?
* as option for game?

My personal point:
I am neither for nor against an option, but if we implement an option, I would prefer the first option, followed by the third. I dislike the second, for the reasons mentioned in #25

**Comment by bunnybot** [comment]
(by csirkeee)
@Astuur (#26): Well, the mapmaker could still control it by placing a tree-type that makse dense forest.

Also, I know that trees spread by seeds :), it's just that most of them die in a very young age. Many times because they don't have enought light, or enough nutrition in the soil, which could both be influenced by having other trees nearby. (Of course I know these trees don't seem large enough to have an effect more then 2 units away, but whatever =) )

Anyway, if this idea is not liked then I'll agree with Nasenbaer in the end of his post (#27) =)

**Comment by bunnybot** [comment]
(by sirver)
I am back and read through the comments. I set this to Won't fixed now. My summarize:

The proposal made a lot of discussion as I expected. The discussion was even on the edge of slipping away from objectiveness. The main opinions were

1) Tree spreading is destructive for fairness and and a deterministic ruleset (on which map makers rely) and should be removed
2) Tree spreading is nice for the aquarium effect and non deterministic variations that it adds to the game and should therefore stay. Most of these opinions agree that fairness and deterministic behavior is skewed but that the effect is too minor and secondary to how they play widelands and want widelands to be played.
3) Like 2, but seeing the arguments in 1) as valid in times. Most of them proposed an option for the map maker or the player to enable or disable tree spreading.

Most people were in favor of either 2 or 3, so the tree cycle will stay. There will not be an option, neither for the map maker nor for the player for the following reasons:

1) Widelands - The Game should have one ruleset on which players and map makers rely. People will not get accustomed to a game were every little bit can be tweaked. We are making a game, not a text editor: One game not many variations.
2) Many options introduces many bugs. Not now maybe, but one day. A nice example of a minor option which I forgot about and which screws up the tutorial is 656224.
3) Widelands is already too complicated. Learning it through tutorials and scenarios and single player games and playing it online should not require a diplomas degree. There is no such a thing as a Widelands Poweruser - all our design decisions must keep the casual player in mind: Adding even more options to the starting of a game will loose us players. KISS - Keep it simple, stupid.

The feature creep is something most open source projects are subject to. I feel one part of my work as chieftain is to keep the feature creep out of widelands and to strive widelands towards becoming simpler, not even more complex.

**Comment by bunnybot** [comment]
(by sigra)
Nasenbaer (#13)
> Sirver (#11)
> > First of all the point if we both have 3 trees at the beginning, one of mine dies and one of yours reproduce, you have build materials for one building more than I do - and wood is so very scarce at the beginning of the game. 
>
> seems to be invalid for me. The first dying/spreading takes place after 800 - 2500 seconds (depending on the type of tree) - so let's take the worst case - all 3 trees are birches (idle for 800 seconds) - so you have got 13 minutes and 20 seconds to build up a wood economy until one of the trees possibly die. - that is more than enough time. And once you have a forester, one tree more or less definitely does not matter, as tree reproduction through the forester is quite fast.

That was how it worked initially. But then it was changed so that the immovable idle times are random. (Binominal distribution with the given time as expected value. So the idle time of a birch would be 1 plus the sum of 2 rolls with a dice from 0 to 799.) This was necessary so that not all trees that were in the map initially (and their decendants) would be synchronized. So actually it is possible that all trees around a player would die before he has built up a wood economy. But that is about as likely as all initial animals running away before he has built up a meat economy. And there is no suggestion how to change that. (If it really has to be changed to give a dying-free time at game begin, it could be changed to for example 721 + the sum of 2 rolls with a dice from 0 to 79 for a birch.)

Kristin (#20):
> Post #16 implies that these are not the true motives of this bug report. It implies that the reasoning contra the tree life cycle is in reality a search for excuses for bad playing. (At least, I cannot understand it otherwise.)

Try to understand it as it is actually written. What I said is that this discussion reminded me of a nice feature from another game. (Would have been better as a separate feature request.) I can understand that the motive for this discussion is a desire to optimize for fairness by minimizing randomness. This discussion comes up once in a while, and the conclusion is that it is not possible to eliminate randomness completely. For example: What if 2 equal soldiers meet? Where should a tree seed land? Where should an animal go?

SirVer (#29)
> There will not be an option, neither for the map maker

That makes sense. It would be too confusing for the players. If the map maker needs more control, more terrain types could be added, where trees do not grow. Then the player can at least see the difference and know what to expect. If the terrain type limit is lifted from 16, that would be a good opportunity to merge the worlds. Then there could be desert regions and greenland regions, blackland regions and winterland regions in the same maps.

**Comment by bunnybot** [comment]
(by ad-simplypeachy)
Having reported bug 1230255 I thought it may be a good time to open this debate again. I have read through this thread and compiled the pros and cons:

Pros:
Adds realism
Makes game more interesting
Adds another economy strategy
	Dealing with trees means expanding your territory has another challenge
Adds ambience
	Similar to animals walking around it is seen as one of the visual appeals of the game
Adds variety to maps
	The next time you play a map it may be different

Cons:
Removes freedom from map designers to decide how the map is laid out
	Affects competitive play
	Affects the intended flow of expansion in the map
Affects tutorial missions
	If player takes too long to reach an area they may not be able to proceed
Affects computer player
	May not take steps to reduce large number of trees
Feature creep
Not noticable by some
	Some feel that it is not a feature which is very noticable until they hit a very large forest - which could have simply been planted there as static by map designer

Of the cons, the following suggestions have been made:
	Tutorial missions need trees removed between stages
	Add terrain types to prohibit / reduce tree spreading
	Add feature to globally toggle tree spreading via a script
	Add multiplayer game feature to globally toggle
	Add another tree type that doesn't disappear after dying to work-around terrain with low tree affinity
	Restrict number of trees that can grow as a result of spread

I had an interesting conversation with SirVer today, having mentioned the port space bug. I said that I liked the tree growth feature but then realised that actually, in my opinion, it doesn't add anything positive to the game. For some extra ambience, realism and "strategy", the game suffers from feature creep, map designs being affected and the fact that some players do not notice it, or consider it a negative feature.

I have realised that it is a feature that I like in theory. A natural, dynamic feature that adds some spice to a game. Something you didn't see last time. But the positives seem so insignificant compared to the negatives that I have to say that I think it should be removed. I notice it when it gets in the way - I have a huge forest to hack away only wastes my time, and is a temporary challenge, I don't notice it after I've removed it. Trees pop up in my farm land so I just leave a few lumberjacks around and then, again, don't notice the feature. I explore to an area which has a few trees there. So what? I cut them down, remove the lumberjacks and don't even notice.

What I notice is what challenges me: narrow areas for roads, difficult-to-defend parts, areas with sparse mountain resources. Pinch points where players battle to expand. These are under the control of the designer. These make the *good* challenge.

If the trees are put there by the designer as a feature - be it for resources, a barrier or as some spice to the map - then it is their choice. The list of additional options, workarounds and extra code to reign in this feature is something which adds more complexity and problems than the original feature gives: some random extra trees.

All this trouble is not worth some random extra trees.

**Comment by bunnybot** [comment]
(by sirver)
>  Trees pop up in my farm land so I just leave a few lumberjacks around and then, again, don't notice the feature.
Notice that if you kill all trees, they will not randomly appear. Only already standing trees will spread over time. 

I agree with ad_ fully here (no wonder, I was fighting for the removal of this feature ever since it was added). I want to add my comment to bug 1230255: 

--- snip --- 
Also, what happens if you land (with a ship) on a port space that is overgrown with trees? You can build the port (there has been special code added in to clear immovables from these places - this code would not have been necessary without the tree spreading), but you might have zero space to build houses. We could add in logic to clear more room around the port, but depending on the terrain, it might be that there is still no space. 
--- snap ---

So to summarize: we added special code and configuration to terrains and bobs to enable this feature. Then we added special handling for the placing of immovables to the game so that scripting does not need to kill randomly spread trees all the time when placing a road or a building. Now we add special code for ports - all for this feature. And here is my personal challenge: Point me to one situation in the replays in the currently ongoing tournament where the tree spreading feature had a noticable non-negative effect to the game play.

**Comment by bunnybot** [comment]
(by frank-pieper-1)
I won't like to lose Tree-LifeCycle lesser as is.

My Solutions is plain explained  that WareHouse-SiteTypes (HQ,Port,WH)get 2 ProductionSite-Slots 
to house up to2 virtual ProductionSites manual or automatical selected from a List of possible (baisc) Site-Types
 and here it is the especially the Lumberer'S House which need to get supported, that the new Port can send out 
a Lumberer to clean any Field from Trees. 

 Perhaps similar like the Geologists but without Need for a Flag to point to the Tree.
BTW the Geologists should get sendable as well to Fields even there is no Flag.

IIRC Geologists are surprisingly hardcoded into Engine -- you can''T remove/rename them from Tribe
as the Site-Code for Geologists is hardcoded into the Engine resp WareHouse, 
Therefore I can't create a Geologists-Site and and redirect Calls for Geologists to that Site resp free WareHouse
from Matter of sending Geologists.  I advice to don't iterate that Erroe a second Time for Lumberer and Port.

**Comment by bunnybot** [comment]
(by frank-pieper-1)
See Post https://wl.widelands.org/forum/topic/1280/?page=4#post-9582 in this Context written before

**Comment by bunnybot** [comment]
(by ad-simplypeachy)
I think that warehouse-style buildings should be just that. They don't need other buildings to be integrated into them because it is already quick and easy to build any of the simple production sites. If you want to clear the trees around your port, you build lumberjacks.

**Comment by bunnybot** [comment]
(by fredkuijper)
Since the water resources appear to have no effect on both trees and wells, these resources could be used to make distinction between areas with and without spontanous tree growth.

**Comment by bunnybot** [comment]
(by fredkuijper)
Problem is however that trees can also grow on mountains, which are not suitable for water resources. Still it's a challenging idea. It would look great to have a path through a forest, or an empty area where a military site can be placed.

**Comment by bunnybot** [comment]
(by sirver)
The question is why would you do that? When the feature is not noticable till after the fact (i.e. you have expanded your territory to see the final result standing there) you might just as well let the mapmaker decide. Or am I missing something?

**Comment by bunnybot** [comment]
(by fredkuijper)
It also effects the areas that are visible, and I like the dynamic of growing trees. But it was not my intention to reopen the discussion, in fact I prefer the current behaviour. I have had no problems with tree growth on the maps that I created. Perhaps it would be an improvement to suppress growth around roads and buildings, but this also costs cpu speed.

**Comment by bunnybot** [comment]
(by hjd)
As I mentioned above, I like the feature and think it puts a nice, unique touch on every game. 

However, due to ports, I have changed my mind. It didn't bother me before seafaring, but not being able to conquer a new island due to random chance doesn't seem right. (Being blocked by an opponent's forester is fair game though ;)) 

It was suggested to have a lumberjack run around and clear the area before constructing the port. Seems ok, but it can get complicated since we have no control of the trees and where/when they start growing. Theoretically, you could end up with a lumberjack running back and forth within the same area because new trees keep popping up, unable to clear the space for your port. Unlikely? Yes. Rare? Definitely. Possible? Unfortunately, yes.

While I'll miss the spreading trees, I think we should remove them early in the build19 cycle to address the problems they cause and see how the game turns out without them.

**Comment by bunnybot** [comment]
(by sirver)
> It also effects the areas that are visible

Yes, but harly noticable. Not al all if you have a forrester or a lumberjack close by. If you can, please provide me with an example where you noticed it in a positive way in the game (ideally in a replay).

> Perhaps it would be an improvement to suppress growth around roads and buildings, but this also costs cpu speed.

That is yet another special casing for this feature - and for me yet another argument that its design does not fit the game.

>While I'll miss the spreading trees...

Why? Is this emotional attachment to a feature that sounds great but isn't or is there something that will be noticeable absent (not trolling here, I am genuinely flabbergasted why so many people like the feature and can just not understand the reasons for it).

**Comment by bunnybot** [comment]
(by fredkuijper)
"Is this emotional attachment"

We are all emotionally attached, with the one or with the other option.

If we clearly desire the complete opposite behaviour and if we are unable to agree at any level, then I suggest to make a switch in the game menu. That would please everybody and hurt none.

**Comment by bunnybot** [comment]
(by fredkuijper)
"where you noticed it in a positive way in the game"

The reason why I like the tree cycle is because it looks more realistic to me. And landscapes always get overgrown when areas are uninhabitated. At the same time it poses no problem in the game if the map designer is careful with the terrain and I can see no reason to change it, except for what hjd noted in #40, but that can be solved in many ways.

"but not being able to conquer a new island due to random chance doesn't seem right"

Does this only apply to MP scenario's? If so, then the scenario can clear the region.

**Comment by bunnybot** [comment]
(by ad-simplypeachy)
fk said:
> "At the same time it poses no problem in the game if the map designer is careful with the terrain"
> "Does this only apply to MP scenario's? If so, then the scenario can clear the region."

Bug 1230255 was on a normal map and it was a problem in the game. The map designer was careful with the terrain and did not place trees in a way that over-crowded a port space. When I arrived, the port was unusable because of the trees.


> "If we clearly desire the complete opposite behaviour and if we are unable to agree at any level, then I suggest to make a switch in the game menu. That would please everybody and hurt none."

What if two players play together, one with tree growth and one without? What if I have tree growth disabled and watch a replay? One of the arguments is that this feature, which doesn't create a strong technical or aesthetic addition, causes a lot of new problems.

If you had the choice to switch off tree growth in a game it would hardly make any difference really. I suspect that everyone would enjoy the same amount of immersion and enjoyment. Which, if a true statement, is a very strong argument against tree growth.

**Comment by bunnybot** [comment]
(by fredkuijper)
"Bug 1230255 was on a normal map and it was a problem in the game. The map designer was careful with the terrain and did not place trees in a way that over-crowded a port space."

I may have misunderstood the problem because I have not been able to do seafaring with expeditions yet. It is new to me that on that map, that I designed, people can do expeditions. To say designed is in this case not the complete story. ;) Is this always possible in MP?

"What if two players play together, one with tree growth and one without?"

That should be done on a per game basis, or set in the map itself.

I start to understand why some guys want to remove the tree cycle, but let us remain thinking productively and with an open view. There are also voices for seefights, and for burning trees, which could also solve this problem. But if we really get stuck in the end with spontaneous tree growth, then we should remove it, without any tears or whatsoever. There would be new features to enjoy.

**Comment by bunnybot** [comment]
(by wolfsteinmetz)
Warning, long posting!

I have been reading through all this now, finally.
In spite of _aD's brave and helpful effort to summarize the argument, I felt
I had better do that.
It was the right idea, because doing so made clear that this isn't just any
mutual exchange of arguments about any old feature.

It's a strive about "the" Widelands paradigm - about the "what WL really stands for"
and naturally emotions run high with this topic, even though we all try to hide this fact.
I might really have missed this insight, if I had just worked on the sober Pros and Cons.

So I will not try to summerize the arguments. Actually I couldn'd have done better than _aD, anyhow, but rather try to explain those different perceptions of the game, and so (hopefully) come to the emotions, also. 
It sure is time the differences were clearly on the table.
I will concentrate on my own emotions, even if it makes me vulnerable, simply because they are the one's I can be sure of, while the sentiments of others nearly always imply some amount of speculation.
I honestly intend no disparagement of anybody's standpoints or feelings, so please give me a second chance to explain before you flame, should you feel offended.

But now right into it without further ado:

Widelands has, like settlers, always attracted different personalities, who sought different kinds of pleasure in this game. And though this certainly is a smooth gradient, the extremes can be categorized into 2 main groups.

There are those, who primarily seek the competition in the game. Let me call them the "Fighters", just for the sake of a handy term.
They enjoy developping their very own sophisticated strategies and tacticts to win this game - according to whatever winning condition, give a very high attention to fairness (player vs player), and welcome predominantly such innovations that offer useful options or alternatives for the military.

I don't think I need to explain this attutude much. Competition is at the base of most any game, and though Widelands offers a significantly different challenge than most other games, the psychological driving force of these players is a very common one. Supremacy over their opponents.


The other kind appriciates more the assembling approach of Widelands. They enjoy the God-mode that WL grants over your territory, and are more preoccupied with optimizing their own economy than with constant glances to their borders. They enjoy seeing that their realm is well funtioning, the productivity is maxed out, and would rather see the inevitable enemy assault postponed until everything is well prepared. They are more prone to look at the animations (or in fact any sort of eye-candy), follow the transport of a particular ware, and are often testing alternatives, and realize the lack of some feature, that would help. Often they prefer larger maps and longer games.
All of this, simply because they are playing the game less time-critical, often alone against the PC, and in my case, sometimes without any opponent at all. Yes, I personally clearly belong here most of the time:)
All in all their attitude is more one of a benevolent king that that of a warlord and they take joy from the "wooseling" and "cuteness" of the game.
I'd like to call them the "Optimizers".

Now these are archetypes of attitudes, not necessarily of real persons. Any player may flip from "optimizer" to "fighter" and back at any time. I only wanted to show the differences.
I have tried to characterize the Optimizers a little longer, since that attitude is alien to some.
Also some fighters despise that attitude. I have been told (not here!) : "WL without enemy?... Go watch "my little pony", you pussy!". All the same, I insist that this too, is a valid way to take joy from this game. It is a bit under-represented, as not all optimizers venture their coming-out.

The conflicts that arise from this dualism range from screen resolution, to map-sizes - and now the tree life cycle. Yes, it's again the same conflict. 
Do I need to mention, that, what you notice, or not notice while playing the game, is largely depending on this?

Some argue, that auto-spreading trees don't belong into this game, because it has the potential to bring unfair advantages to some player, thus making "winning" a random chance.

I'd like to provoke a bit at this point.
What do you think of this thesis:
"The whole fighting and military business is an alien element in WL."
It doesn't belong to an economical simulation, and warfare should never be an option!
It has only been implemented to satisfy the "fighters".
Place all opponents on a map. Each one on his own, identical island (no seafaring) and let the statistics (including military strenght, if you will) decide who has won.
We could compute a new GNP and make that a winning criterium.
See what I mean?
I personally would not miss much in such a setup, others would.


_aD wrote: 
>>Adds ambience: Similar to animals walking around it is seen as one of the visual appeals of the game

True of course, but a bit too weak a statement. Animals (and fish) are not autoreproducing like trees are.
So in fact the trees, or forests rather, are the only items that show some evolution in Widelands (independent from player action).
And they are also the only elements that react to the environment that they are placed into.
By "react" I mean, that, given enough time, you will see that the different tree types spread according to their terrain affinity and supersede other tree species, that grow, or spread slower on a give terrain.
This makes them a unique feature (Settler never had it) and also in my view not really comparable with the relatively simple way the fauna acts. There once was the idea of auto-breeding animals and fish, too, but it has been turned down, due to counteracting the unique abilities of Barbarian gamekeepers and Atlantean fishbreeders.

It really is, what gives WL "a life of its own", adds credibility etc. and it's the link between economy and ecology.  But I need not repeat that all.
I agree, that these tree features could be better visible, though.

And now you come with this idea of getting rid of the whole tree-growing system!.Oouuch:)
Why would you want to this? Simply because it's a nuisance sometimes?
Sigra has thoughtfully made this system very configurable. I can be tweaked.

Doing away with it would totally ignore the preferences of the Optimzers.
Instead of this I had envisioned that some day I might be able to convince the devs, that expanding the system of terrain affinity to wheat, corn, blackroot, reeds etc. would add a lot to the game.
Not in the sense of "nice eye candy" only, but also for tactics and strategy. 
If I had my say,I would like to make the game a _lot_ harder to play in the sense of placing decisions.
Windmill on a mountain ridge? Wise move - more flour.
Vinyards on the south slope? Clever, more grapes.
A farm in the midst of Smelting works and Charcoal burners? Dumb decision, less wheat.
And so own... you get the picture.
It could make for some very interesting gameplay.

Yes, a different game, may be ... it will not happen, I know.

Sorry for digressing with this, it was too tempting.

Back to trees
It's all a question of how much effort you want put into it.
There is really no reason to remove the tree cycle and make the optimizers unhappy.
What really is annoying at times, is that all forests become impenetrable.
No way through and no place to set a lumberjack, except at the borders.


For this general overgrowth problem, I can see a simple solution, maybe.
Create an extra tree species "Tree0".
Make it non-blocking like the dead trees and invisible.
Adjust its parameters, longevity, terrain affinity and seeding so that it is competitive to all other trees.
You can now adjust the penetrability of a forest by tweaking the lifespan of this Tree_0.
You may need some other adjustment, so that lumberjacks do not cut the invisible trees and foresters don't plant them.

As for the port space problem, I've already made a suggestions at bug #1230255

Generally speaking, I would like to suggest to give the mapmakers
more control over the trees.
A generally modifying factor, so that tree seeding activity goes from 0 to max,
plus control over the tree_0 feature (allowed/not allowed) might be enough.


Please don't touch my trees! :)

**Comment by bunnybot** [comment]
(by sirver)
I try to reply short to not keep this thread from exploding. 

>  this isn't just any mutual exchange of arguments about any old feature.
I disagree. This is just another feature - I feel that there are a lot of emotions around it, but in its core it is just the question: is it good for widelands or bad.

[fighters vs optimizers]
> Doing away with it would totally ignore the preferences of the Optimzers.
I can buy into this concept of categorizing the players. But Kristin (for example) would be a very clear Optimizer - and she hates the tree cycle too (she is also a very skilled map maker, which might also feed into this). So this is not a strong argument for me. Besides - optimizing seems to have to do very little with if trees are already there when you reach a certain point in the game later rather then earlier. Fighting trees is not much optimizing. 

["The whole fighting and military business is an alien element in WL."]
Yes - I always made the point very clear. Widelands lays the focus on building up, not ripping down. That is why the military is very economic focused too - if your economy works well, your military is strong. I tried to design the game to reduce military even more, but to this point, it is still the sole point of interaction between the players. The game you describe, without military, means that you are playing alone - even in network games, because there is no interaction at all. This is hardly a multiplayer game at all.
Tangent: I personally love collectors for the very thesis you wrote down there - military and warefare consume the wares that give the most points, so you actually have to balance protecting yourself and hording wares that give you points. This emphasises a strategic concept that takes away from conquering and adds an economic parameter while still allowing interactions between the players.

> It really is, what gives WL "a life of its own", adds credibility etc. 
I challenge this conclusion. Widelands life of its own is in the economy and its interaction which is plenty. The tree life cycle has no impact on your inner economy - you never notice it in areas you owned for more than 15 minutes - I looked at all the replays from the widelands tournament so far and saw many a game and I find this to be a true statement everywhere (please, proof me wrong). You only see it outside your territory - and this might have been designed by the map maker just as well.

[Tree0 concept]
I think you mean it in a different way, but to me this reads like suggestion to remove the tree spreading feature in some places - through a new concept. Inpenetrable forrests are a map design aspect - they can be great (I remember a S2 map where you where separated by your enemy through a huge forrest - you could choose to cut it down or build more trees to make it harder to penetrate - a nice interplay of map designers decision and players decision - in Widelands the thickness of this barrier would be a function of time you took to expand to this place. The map designer can not work on any assumptions in this area to further design the map really). 

> Generally speaking, I would like to suggest to give the mapmakers more control over the trees.
That is a strange conclusion to me as well - this is exactly what I am suggesting. Let the map designer decide where trees should be when the player arrives.

Thanks for taking the time for writing all this Astuur, I interpret a lot of things quite differently though as you can see.

Gee, this post got very long too.

**Comment by bunnybot** [comment]
(by wolfsteinmetz)
>>I try to reply short to not keep this thread from exploding.
Having explained my general notion already, I will try to do likewise here to save everybody's time ;). 
Can't make promises, though...you know me anyway :)

>>I feel that there are a lot of emotions around it, but in its core it is just the question: is it good for widelands or bad. 

Given the different thrills that people find playing WL, it will always be more like. "Is it good or bad for _my_way of playing Widelands". Guess, we need to accept this.

>>The game you describe, without military, means that you are playing alone - even in network games, because there is no interaction at all. This is hardly a multiplayer game at all.
True, I said I was provoking with this ;) It served only to make clear that sometimes you have to compromize in order to have a game,that can please more players.
Sorry, if I made you write more than necessary; I know that warfare WL-stye _is_ needed to somehow reap what you have sowed with your economy, and I think WL does a good job at that.

But here is a point that itches me:
I'd like to reverse the principle argumentation of the discussion!
"Tell me what you hate about the tree life cycle." instead of "Show me, where you see any benefit"
Why? 
It don't agree that an existing and working feature, that is liked by many, should have to justify its existance. It should not even matter, why people like it. They simply do. Anyway it's easier to explain annoyances that you have experienced, than to imagine the shortcomings of a world without the feature.

And from this standpoint, we should ask:
"How can those annoyances be fixed?"

This seems to me to be the route this discussion should follow - Not vice versa.
The opponents of the feature do have good points in the argument and I share some of them.
If fixing the annoying points is too cumbersome, or no programmer is willing to work on solutions,it must be dropped. But that is the ultima ratio.

Let me clearly say, that I too have my own points of criticism towards the tree-cycle _as_it_is_. But my stance is to improve the feature, not remove it. It urgently needs tweaking for terrain-affinity, spreading radius and an attenuation of the exponential way trees multiply. (I'm currenty testing values for some of these)

Nobody has up to now said what exactly should be removed, but judging from the arguments  the 8 species and their terrain-affinity is not critical.
By the way - this is what you_do_ notice; that different terrain produces different woods with other tree species.
(Should be optimized, too). 
Among all the properties that those trees have, the only critical one seems that they seed on open terrain and so inevitably slow down the enlargement of a player's territory, and that they re-occupy already cleared space that you want for farms, vinyards and such.

Let's start by the latter:
The main consequence is probably, that you need to keep some evenly placed lumberjacks throughout, as long as you do have some trees near. (natural or planted) to keep your farms, vinyards etc. clear. 
I quite like that requirement; it's reasonable and not asking too much. 
I like the eye-candy part of it as well. A landscape with some few half grown, scattered trees between the houses looks far better, than a completely treeless one.

Now to the forest overgrowth.
I have learnt to hate that, too, as both, a map maker and a player.
Can we all agree, that the woods might be tolerable, if they were not so thick set?
If there was no problem to place a lumberjack or a sentry and connect them inside the forest ?
I would be content if this was the case as a player. These sparse forests would not hinder a player's progress and still the trees would show their natural behavior.
I am currently testing something along that line, in the primitive way that I can. I could do better. if I found some documentation for the statements allowed inside tree conf files.(hint! hint!)

The principle idea, is that of "tree0" that I have failed to explain well enough in my last posting.
So let me try again here. I would like to introduce a 9th tree species into the game. Let's call it "tree0". 
It is an ordinary tree, that does anything that all other trees do. It can be placed by a mapmaker in any number wherever he/she likes (or be not used at all). It will grow, and seed. 
There are only two difference compared to all other trees: They are invisible (their png is totally transparent), and they do not block roads or houses. They do block the space for any other trees though.
In this respect tree0 behaves like the "tiny" version of all trees, or like a dead tree. 
I would give them a terrain affinity of 254 for all terrains, an insane long lifespan, and only a marginal seeding radius of 1 or 2, so that their siblings replace more or less just their fallen anchestors.(tested, looks good)

The task to inlude this species into the game editor, to disallow it to be planted by the rangers, and to make it unrecognizable for the lumberjacks remain however, and are above my head.

So - for the mapmaker.. if you want to keep your harbor space accessible, you place some tree0s there.
If you want to keep some other passage or area clear from tree growth, again place some tree0s there and you are done. 
The other trees will overgrow, but will have no chance to occupy the space that you have reserved with tree0.
You want a map with only very few trees, but still want a terrain useful as farmland?
Scatter enough tree0s, and some others - done.(so far untested)
... and there will still be all other benefits of the tree life cycle going on as usual in the areas that you did not place the tree0s.

The rationales to suggest such a solution is, that even a mapmaker does not need full control over the tree growth on each an every coordinate of the map. He may need to forbid growth in certain areas, but in other places the tree cycle can go on as usual.

Found any flaws in that idea? 
Feasable?
_____________________________________________
When I said, that I wished mapmakers had more control over the trees, I had already envisioned the above to be implemened, but I had some further, not very ripened ideas.
The tree0 helps to avoid certain problems, but it isn't what I'd call control over trees. 
The most powerful solution was, if we could make all tree properties from the conf file configurable in the editor - seeing the conf files as the default, but one that can be modified for each map from within the editor.

If this could be achieved, we would of course also have configurable (on a per map basis) properties for tree0, opening up for a more dynamic setup of the open spaces, and (for the first time) scenarios with trees slowly decreasing.(by tree_0 overgrowth or direct modification of there properties) Things like the "birch-pest" could become possible.

I like the idea that map makers can surprise the players, so I don't accept that they should have a clear idea about what to expect.
They need not.

**Comment by bunnybot** [comment]
(by wolfsteinmetz)
Here is a screenshot from an experiment. to give you some impression of the possibilities.
We are 28h into the game and the original map had an evenly mixed, dense patch of all 8 tree species, the size of a ranger's working area, placed in the middle of all terrain samples.
The tree paramenters are modified here already for all trees with the intention to prevent an overgrowth.
So no tree on this map is behaving the way you would expect from your experience.
Tree8 has been made are non-blocking and given a terrain affinity of 222 for all terrains.
Tree8 specimen set by the map maker and their siblings are filled white squares.
Tree8 speciment planted by the ranger are diagonally striped. They do not seed.

Please notice how tree8 is unrivalled on wiese1, while on bergwiese the larch is a superb competitor (terrain affinity of 232)
Also please mark how the tree8 placeholders do not block, and the general sparseness of the wood.

**Comment by bunnybot** [comment]
(by wolfsteinmetz)
Just found an additional problem.
Protecting the port space is not quite as easy.
Contrary to the "normal" big, medium and small buiilding places, a port space must be explicitly set by the mapmaker and is an item that connat co-exist with any other object on the same coordinate.

**Comment by bunnybot** [comment]
(by janitor)
[Expired for widelands because there has been no activity for 60 days.]

**Comment by bunnybot** [comment]
(by flaviojs2005)
I started playing Widelands recently. I noticed the tree cycle when I tried changing the speed to 10x (lazy way to get more resources before getting to the combat map) and soon found trees everywhere. xD
- Maybe make a note of it in the lumberjack part of the tutorial?

I wasn't sure what to think of it but after playing some more I now feel that it adds depth to the game, increasing strategy complexity, since we have to account for the unpredictability of nature. I'm for keeping it.

It makes the game feel more real, however it seems to be lacking in a particular detail: human impact
In the real world, having humans going back and forth in a trail is enough to decrease the chance of big vegetation like trees being able to grow around it. (we have pretty destructive habits)
Some suggestions:
- a tree seed could have a chance of not taking root based on the recent activity of the 6 neighboring cells (activity would be a working building or goods being transported in a road or a worker passing by the cell; inactivity would be a stopped building or an idle worker in the road or a normal cell)
- a tree sprout could have a chance of dying based on the activity of the neighboring cells

**Comment by bunnybot** [comment]
(by mquinson)
I second the suggestion of Flavio: the tree life cycle would not be anoying if it just added some trees without covering the map. Just make sure that no area gets flooded with the trees, and you have a very nice feature.

**Comment by bunnybot** [comment]
(by ad-simplypeachy)
"adding some trees without covering the map" can be done when the map is first created. Tree growth isn't required for this.

**Comment by bunnybot** [comment]
(by sirver)
I believe that there are ways to make the tree life cycle less or not annoying at all. My argument still stands though: it does not add to the game, at best it is not annoying. I keep asking for proof where somebody noticed the tree life cycle in a game in a positive fashion and so far nobody could show me a reply or even an anecdote.

**Comment by bunnybot** [comment]
(by gunchleoc)
Maybe we could keep some of the visual attraction without having to deal with the problems. If we completely remove the feature, once forests are fully grown, they will stay static.

So, I was thinking, how about every mature tree spawns one and exactly one new tree, irregardless of terrain affinity. This would keep the number of trees stable, while still showing trees in various stages of growth.

However, since trees would shift their positions, we might still have problems with port spaces. So, how about this: Every tree that dies a natural death (= not being felled by a lumberjack) respawns in its location (maybe make the interval a bit random). This way, we will have trees in all stages of growth on the screen, while resolving all problems for map designers.

**Comment by bunnybot** [comment]
(by ad-simplypeachy)
I do not play Widelands because it is a visual attraction. Don't get me wrong - I think the game is very pretty, but I play it because of the challenge of making economies and the strategy involved. I don't play Widelands to look at forests. If I wanted to look at a forest I would build a forester's hut, leave it for a while, then dismantle it.

As with most of the other suggestions, single-tree-growth suffers the same way: it does not address the problem satisfactorily. "However, since trees would shift their positions, we might still have problems with port spaces."

This weekend I gave up playing one of the newly-posted large maps because I got sick of building dozens of lumberjacks for every screen of new territory I found. It got boring and tedious once I was expanding in three directions at once.

**Comment by bunnybot** [comment]
(by wl-zocker)
I agree with #57: Most people play Widelands to build up an economy, and not to fell hundreds of trees.
While I like logical games, too much naturalness is not always good - especially when it hinders the gameplay. The tree life cycle is such a feature: I love to play larger maps to be able to build up a good-working economy; but when expanding, there are often a lot of trees that slow down my expansion. Even when all trees are cut down, I still have to leave some lumberjacks' houses to avoid the trees recover my area. Apart from this function, they are of no avail, and it is quite annoying to always have to keep an eye on them (Do I have to build more? Can I dismantle some lumberjacks' houses?). I therefore think the tree life cycle does not make the game better, but rather scarier (Where do all those trees come from?).
Trees that overgrow port spaces are another problem: They hinder the player in discovering the map. They can even destroy the idea of the map (how the mapmaker wants the game to develop) and therefore limit the mapmaker's freedom. I am not sure if specialcasing for port spaces is a good idea - we do not know what future features will look like.

**Comment by bunnybot** [comment]
(by ad-simplypeachy)
https://wl.widelands.org/maps/full-moon-11/
Author of the map fk states in the map description: "The number of trees in this map has been reduced significantly [from map v1.0] after it turned out that they were an obstacle during the game."

**Comment by bunnybot** [comment]
(by ad-simplypeachy)
Modified title as I couldn't find the bug when searching for "tree growth" (and spammed the wrong bug report as a result :-)

**Comment by bunnybot** [comment]
(by fredkuijper)
#56 "If we completely remove the feature, once forests are fully grown, they will stay static."

"Every tree that dies a natural death (= not being felled by a lumberjack) respawns in its location (maybe make the interval a bit random)."

I agree with that, as it deals with the problem most profoundly. I think that everyone would be pleased with his proposal.

**Comment by bunnybot** [comment]
(by sirver)
Setting to incomplete for bug sweeping.

**Comment by bunnybot** [comment]
(by janitor)
[Expired for widelands because there has been no activity for 60 days.]

**Comment by bunnybot** [comment]
(by tiborb95)
After last changes related to terrain affinity trees stuff needs serious attention, because a lot of map are unplayable because they are not able to grow trees in reasonable speed. So this is apposite problem to this bug.

**Comment by bunnybot** [comment]
(by franku)
My emotion says: Its a nice feature...
My logic says: It make more problems than it is usefull... so remove the life cycle of trees

I don't know if this feature is really a great "visual attraction". I think most of the players do not even get aware of it, except they are stumbling of (f.e.) "suddenly growing trees on my farms".

Another thing is: A forest do not spread in such a huge way in a short time. Since wiedelands has no "time" (except the time of gameplay), a fast spreading forest makes no sense (somehow, i couldn't explain exactly what i mean).

The proposal of #57 is maybe a solution. But i think it's more importand to get the terrain affinity to work in a good way. Until then the lifecycle of trees should be removed and, after some time, we could see if that is really a "must have".

**Comment by bunnybot** [comment]
(by lainluigi86)
about tree terrain affinity, that should have been just addressed. 

About the rest, a possible solution would be to activate the tree growth only when a (human) player has explored the area. So, if you planted a few trees on the map, the player is still going to see a few trees, which will then start to grow and fell.
Personally I like the feature, even if I would not die if it was removed. But would be a pity, with all the time people took to make it work. It makes forests look more real, with holes in it and trees shifting. I generally do not play large maps, but when I did, I never noticed the problem people are complaining about. sure, there are some trees growing where you want to expand, but it's not THAT bad.

**Comment by bunnybot** [comment]
(by dragonatma)
Hmm...

If people are worried about clearing out overgrown forests as a hassle, why not add a new worker/building/ware, "Clearer"/"Clearer's hut"/"torch"?

They'd have a huge range (maybe 15-20 tiles radius) and would certainly work faster than lumberjacks, but instead of leaving logs they'd simply burn down the trees, leaving no wares.  I assume their AI would be something like this:

[loop start]
Go to nearest tree
Burn down tree [animated flame over the tree for a few seconds, then the tree and flame disappear]
Go to nearest tree
Burn down tree
Go to nearest tree
Burn down tree
Go to nearest tree
Burn down tree
Go to nearest tree
Burn down tree
Return home
Wait ten seconds
[loop end]

And for those who are worried about trees overtaking their farms in a non-frontier area, just do what I do: a few foresters densely packed, and a loose ring of lumberjacks around them.  Unless the lumberjacks' ring is *too* loose, they work well at keeping the trees contained.

**Comment by bunnybot** [comment]
(by gunchleoc)
A clearer would still not solve the problem with the port spaces. I think all other problems have been addressed now?

**Comment by bunnybot** [comment]
(by franku)
Yes, i think this bug could marked as solved. Feel free to reopen this again :-)

**Comment by bunnybot** [comment]
(by gunchleoc)
Fixed in build19-rc1.
